highlight_view.py

```diff
@@ -107,37 +108,25 @@ def get_base_error_style(linter, code, error_type, **kwargs):
     return store.get_style(code, error_type)
 
 
-def filter_errors_for_gutter(errors):
-    errors_by_line = defaultdict(list)
+def filter_errors(errors, group_fn):
+    grouped = defaultdict(list)
     for error in errors:
-        errors_by_line[error['line']].append(error)
+        grouped[group_fn(error)].append(error)
 
-    # Since, there can only be one gutter mark per line, we have to select
-    # one 'winning' error from all the errors on that line
     filtered_errors = []
-    for line, errors in errors_by_line.items():
-        # We're lucky here that 'error' comes before 'warning'
-        head = sorted(errors, key=lambda e: (e['error_type'], -e['priority']))[0]
+    for errors in grouped.values():
+        head = sorted(errors, key=lambda e: (-e['priority'], e['error_type']))[0]
```

Comment:
Here, we just flip priority and error_type

Reply:
For me, this is kind of a bug fix, bc I actually tried to 'promote' a warning over an error, which never worked. So now I know why.